classes/ActionScheduler_AdminView.php

```diff
@@ -35,9 +35,42 @@ public function init() {
 			} else {
 				add_action( 'admin_menu', array( $self, 'register_menu' ) );
 			}
+			add_action( 'admin_notices', array( $self, 'check_past_due_actions' ) );
 		}
 	}
 
+	/**
+	 * Check if there are any past due action
+	 *
+	 * This method will check if there are past due. Because Action Scheduler is action agnostic,
+	 * it will execute an action with a list of the actions.
+	 *
+	 * The action owners should be listening to this action, and do something to warn the users
+	 * about how to fix their WP-Cron or to run their actions manually.
+	 *
+	 * This method is executed by the `admin_notices`, so it is safe to render the admin notification
+	 * from the `action_scheduler_past_due_action` action.
+	 *
+	 */
+	public function check_past_due_actions() {
+		$instance = ActionScheduler_Store::instance();
+		if ( $cache = get_transient( __METHOD__ ) ) {
+			$action_ids = $cache['action_ids'];
+		} else {
+			$action_ids = $instance->query_actions( array(
+				'date' => new Datetime( apply_filters( 'action_scheduler_past_due_time', '-5 days' ) ),
+				'status' => ActionScheduler_Store::STATUS_PENDING,
+			) );
+			set_transient( __METHOD__, compact( 'action_ids' ) );
```

Comment:
Why use `compact()` here when the array only has one value?

Perhaps the transient name needs to be more suitable to avoid that. Something like `action_scheduler_past_due_action_ids`?

Reply:
The `set_transient()` call needs an to be passed the 3rd `$expiration` parameter, otherwise actions will continue to show as past due until the entire transient cache is cleared - the default expiration time is indefinite.

Reply:
Just for my quick prototype so I could check if the object was valid or not. If the prototype looks good in overall I will implement the following improvements:

1. 24 hours of expiration
2. I will cache only the ID, while fetching the objects from the DB I would check if they are still pending. If they are not, I would invalidate the cache (and I would show the notices in the page load).
3. I would somehow limit to just 10 or so elements. If there are more, it is obviously something that needs to be shown like a red warning in the Action Scheduler page.

Reply:
Ah gotcha. That makes sense. Those 3 things sound good. 👍